src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -395,10 +393,39 @@ protected Object readResolve() {
         this.setTokenFieldToCheckKey(this.tokenFieldToCheckKey);
         // ensure escapeHatchSecret is encrypted
         this.setEscapeHatchSecret(this.escapeHatchSecret);
+        try {
+            if (automanualconfigure != null) {
+                if ("auto".equals(automanualconfigure)) {
+                    OicServerWellKnownConfiguration conf =
+                            new OicServerWellKnownConfiguration(wellKnownOpenIDConfigurationUrl);
+                    conf.setScopesOverride(this.overrideScopes);
+                    serverConfiguration = conf;
+                } else {
```

Comment:
consider making this `else` into an explicit `if` to protect against future changes?  if the expected values of `automanualconfigure` ever change and this is not updated then you would get the default case even when you don't want it

Reply:
The field is dead from this version onward so there would be no issues with it changing in future.

Just here for the `readRosolve` and is transient so it's either `"auto"``"manual"` or `null`